pkg/theia/commands/anomaly_detection_run.go

```diff
@@ -151,6 +151,39 @@ be a list of namespace string, for example: '["kube-system","flow-aggregator","f
 	}
 	throughputAnomalyDetection.ExecutorMemory = executorMemory
 
+	aggregatedFlow, err := cmd.Flags().GetString("agg-flow")
+	if err != nil {
+		return err
+	}
+	if aggregatedFlow != "" {
+		switch aggregatedFlow {
+		case "pod":
+			podLabel, err := cmd.Flags().GetString("pod-label")
+			if err != nil {
+				return err
+			}
+			if podLabel == "" {
+				return fmt.Errorf("agg-flow pod requires Pod labels, None given: %v ", podLabel)
+			}
+			throughputAnomalyDetection.AggregatedFlow = aggregatedFlow
+			throughputAnomalyDetection.PodLabel = podLabel
+		case "external":
+			externalIp, err := cmd.Flags().GetString("external-ip")
+			if err != nil {
+				return err
+			}
+			if externalIp == "" {
+				return fmt.Errorf("agg-flow external requires external IP, None given")
+			}
+			throughputAnomalyDetection.AggregatedFlow = aggregatedFlow
+			throughputAnomalyDetection.ExternalIP = externalIp
+		case "svc":
+			throughputAnomalyDetection.AggregatedFlow = aggregatedFlow
```

Comment:
Why we require users input Pod labels and IP for "pod" and "external" cases but not require a service name for the "svc" case?
I feel Pod labels and external IP are optional input, if user doesn't provide this info, we will considering all possible pod labels and external IPs. Like what we have done in the "svc" case.

Reply:
Sure, I can add it as optional parameters, just wanted to confirm if the user doesn’t provide labels or external ip, should we have a check if the corresponding columns are empty or should we even include those columns in DF ?

Reply:
For example if user choosed "external" agg type, we will consider all toExternal flows, group by the destinationIP, aggregated throughput for each destinationIP, and find any anomalies.
Hope this exmaple answered your question.

Reply:
for “external” case, it may work as we have flow type to define if the traffic is for external, but there could be pods with no labels, and pods with any type of label, should we add both of them? they both would have different kind of queries.

Reply:
Currently, we should not consider the empty cases, as we look for pods based of labels, if there are no labels it would be little misleading to still collect them based of their name.

Reply:
What I meant was to have a user input parameter 'podname' that would cover cases where the user knows the pods they're interested in don't have labels. 
Just a nice-to-have.